src/Cogworks.CogFlare.Core/Services/CachePurgeService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using static Umbraco.Cms.Core.Constants.Conventions;
+
+namespace Cogworks.CogFlare.Core.Services;
+
+public interface ICachePurgeService
+{
+    Task PurgeExternalCacheAsync(IEnumerable<int> ids, CancellationToken cancellationToken, string notificationLabel, bool isMedia = false);
+}
+
+public class CachePurgeService : ICachePurgeService
+{
+    private readonly ICloudFlareCachePurgeService _cloudFlareCachePurgeService;
+    private readonly IUmbracoContentNodeService _umbracoContentNodeService;
+    private readonly IRelationService _relationService;
+    private readonly CogFlareSettings _cogFlareSettings;
+    private readonly ILogger<ExternalCachePurgeComponent> _logger;
+
+    public CachePurgeService(
+        ICloudFlareCachePurgeService cloudFlareCachePurgeService,
+        IUmbracoContentNodeService umbracoContentNodeService,
+        IRelationService relationService,
+        CogFlareSettings cogFlareSettings,
+        ILogger<ExternalCachePurgeComponent> logger)
+    {
+        _cloudFlareCachePurgeService = cloudFlareCachePurgeService;
+        _umbracoContentNodeService = umbracoContentNodeService;
+        _relationService = relationService;
+        _cogFlareSettings = cogFlareSettings;
+        _logger = logger;
+    }
+
+
+    public async Task PurgeExternalCacheAsync(
+        IEnumerable<int> ids,
+        CancellationToken cancellationToken,
+        string notificationLabel,
+        bool isMedia = false)
+    {
+        var urlsToPurge = new List<string>();
+
+        foreach (var id in ids)
+        {
+            if (IsKeyNode(id))
+            {
+                _logger.LogInformation($"Full purge triggered: [{id}] Key node {notificationLabel}");
```

Comment:
I'd personally extract lines 45-46, 55-56 and 61-74 into separate private methods. It would not only make the main method smaller but it would also make it more readable as the method names would describe what's happening there before we even start reading the code.

For example we could extract lines 61-74 into a PurgeRelatedCacheAsync or anything among those lines 😉